pkg/controller/kogitodataindex/resource/statefulset.go

```diff
@@ -26,14 +26,39 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"path"
+	"strconv"
 )
 
 func newStatefulset(instance *v1alpha1.KogitoDataIndex, secret *corev1.Secret, externalURI string, cli *client.Client) (*appsv1.StatefulSet, error) {
+	// define the http port
+	httpPort := defineDataIndexHTTPPort(instance)
+	log.Debugf("The configured internal Data Index port number is [%i]", httpPort)
+
 	// create a standard probe
 	probe := defaultProbe
-	probe.Handler.TCPSocket = &corev1.TCPSocketAction{Port: intstr.FromInt(defaultExposedPort)}
+	probe.Handler.TCPSocket = &corev1.TCPSocketAction{Port: intstr.IntOrString{IntVal: httpPort}}
 	// environment variables
 	removeManagedEnvVars(instance)
+
+	// set KOGITO_DATA_INDEX_HTTP_PORT env
+	hasEnv := false
+	for envName := range instance.Spec.Env {
```

Comment:
```suggestion
	port, hasEnv := instance.Spec.Env[DataIndexEnvKeyHTTPPort]
```

There is a better way to get values from a map instead of using a loop, example: https://play.golang.org/p/kHJ9ctbO4UP

Reply:
nice, thanks